services/udf/service.go

```diff
@@ -106,15 +109,18 @@ func (s *Service) Refresh(name string) error {
 		return fmt.Errorf("failed to load process info for %q: %v", name, err)
 	}
 	s.infos[name] = info
-	s.logger.Printf("D! loaded UDF info %q", name)
+	s.diag.LoadedUDFInfo(name)
 	return nil
 }
 
 func (s *Service) loadUDFInfo(name string) (udf.Info, error) {
 	// loadUDFInfo creates a UDF connection outside the context of a task or node
 	// because it only makes the Info request and never makes an Init request.
 	// As such it does not need to provide actual task and node IDs.
-	u, err := s.Create(name, "", "", s.logger, nil)
+
+	// TODO(desa): that okay? Nil check against this happens in udf.go logStdErr()
+	// alternative would be to have a mock implementation that gets passed in here
+	u, err := s.Create(name, "", "", nil, nil)
```

Comment:
I don't think this is okay, its possible valuable information is lost if we do not provide a diagnostic here. Looking at it the simplest approach would be to add a WithUDFDiagnotic method to the Diagnostic interface within this package. The udf.Diagnostic interface is quite simple so its shouldn't be hard to accommodate. Thoughts?